python/src/equistore/operations/join.py

```diff
@@ -82,29 +82,33 @@ def _join_labels(labels: List[Labels]) -> Labels:
 
             properties = ["tensor", "q"], [[0, 0], [0, 1], [0, 2], [1, 0], [1, 1]]
     """
-    names = [label.names for label in labels]
-    values = [label.tolist() for label in labels]
+    names_list = [label.names for label in labels]
+    values_list = [label.tolist() for label in labels]
     tensor_values = np.repeat(
-        a=np.arange(len(values)), repeats=[len(value) for value in values]
+        a=np.arange(len(values_list)), repeats=[len(value) for value in values_list]
     )
 
-    unique_names = np.unique(names)
-    unique_values = np.unique(np.vstack(values), axis=0)
-
+    # We have to flatten the list because the number of elements in each sublist
+    # could be diferent which causes a problem for Numpy.
+    unique_names = np.unique([name for names in names_list for name in names])
```

Comment:
I don't get what this list comprehension is doing. Could you add a short example? Why does `for name in names` appears twice here?

Reply:
It flattens a list of lists

```python
names_list = [('prop_a', 'prop_b', 'prop_c'), ('prop_a', 'prop_b')]
[name for names in names_list for name in names]
```

results in 

```
['prop_a', 'prop_b', 'prop_c', 'prop_a', 'prop_b']
```

we can also use `functools.reduce(operator.concat, names_list)` as alex suggested if this is clearer for you.

Reply:
the functools version is not much clearer for me. But adding the small example you give here in a comment would be nice!

Reply:
I will change this to the functools version and extend the comment.